src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -328,7 +347,7 @@ srv_callback (const char *service,
    if (size < 1) {
       DNS_ERROR ("Invalid record in SRV answer for \"%s\": \"%s\"",
                  service,
-                 strerror (h_errno));
+                 _mongoc_hstrerror (h_errno));
```

Comment:
I'm trusting that `h_errno` is always set by this point. I do find it odd that we rely on it being declared and reference it here even if _both_ DNS APIs used above (which would set this) are unavailable. Or would libmongoc never compile without at least one of those available?

I left some more general comments about using `h_errno` in [CDRIVER-4028](https://jira.mongodb.org/browse/CDRIVER-4028?focusedCommentId=3907963&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-3907963), but that's outside the scope of this PR.

Reply:
Quoting from the [resolver(3) man page](https://man7.org/linux/man-pages/man3/res_nsearch.3.html):
> In the case of an error return from res_nquery(), res_query(), res_nsearch(), res_search(), res_nquerydomain(), or res_querydomain(), the global variable h_errno (see gethostbyname(3)) can be consulted to determine the cause of the error.